src/http-client.js

```diff
@@ -105,12 +105,13 @@ export class HttpClient {
         if (Response.prototype.isPrototypeOf(result)) {
           response = result;
         } else if (Request.prototype.isPrototypeOf(result)) {
+          request = Promise.resolve(result);
           response = fetch(result);
         } else {
           throw new Error(`An invalid result was returned by the interceptor chain. Expected a Request or Response instance, but got [${result}]`);
         }
 
-        return processResponse(response, this.interceptors);
```

Comment:
The only change you should need in this method is to add `result` as a third argument to `processResponse`.